src/test/java/spoon/test/generics/GenericsTest.java

```diff
@@ -960,4 +961,68 @@ public void testClassContextOnInnerClass() throws Exception {
 		//contract: generic types defined in enclocing classe (Banana<T>) are resolved from inner class hierarchy (Vitamins->List<T>) too.
 		assertSame(classBanana.getFormalCtTypeParameters().get(0), new ClassTypingContext(classVitamins).adaptType(refList_T.getActualTypeArguments().get(0)).getDeclaration());
 	}
+
+	@Test
+	public void testWildCardonShadowClass() throws Exception {
+		// contract: generics should be treated the same way in shadow classes
+
+		// test that apply argument type contains a wildcard
+		Launcher launcher = new Launcher();
+		Factory factory = launcher.getFactory();
+
+		launcher.addInputResource("src/test/java/spoon/test/generics/testclasses/FakeTpl.java");
+		launcher.buildModel();
+
+		CtInterface<?> fakeTplItf = factory.Interface().get("spoon.test.generics.testclasses.FakeTpl");
```

Comment:
OK, for a more readable test, it would be great to extract in an assertion and then you would write

```
// normal class
checkFakeTpl(factory.Interface().get("spoon.test.generics.testclasses.FakeTpl"))
//shadow class
checkFakeTpl(factory2.Interface().get(spoon.test.generics.testclasses.FakeTpl.class))
```